quickstart.md

```diff
@@ -33,9 +36,58 @@ INFO[2020-10-15T18:43:27Z] - fixedSizeExampleTest: PASSED
+
+The `IsServiceUp` method will be used to inform Kurtosis about when a particular service instance is actually available to ensure tests aren't run before the whole test network is available. During network setup, Kurtosis will continually call the `IsServiceUp` method until either it returns true or the timeout defined by `GetTimeout` is reached. Note that the service's dependencies are given as arguments to `IsServiceUp` but aren't used here - Kurtosis passes in a service's dependencies as an argument in case our service's availability is contingent on the state of its dependencies, but this won't be used in most distributed systems where nodes simply won't report themselves available until they successfully connect to their dependencies.
+
+### Test Network
+We're all set up to use this service in a network now... but of course, we still need to define what that network looks like. Kurtosis has a `ServiceNetwork` object that represents the underlying state of the test network, but interacting with it is often too low-level for writing clean tests. Just like with services, Kurtosis lets the you define an arbitrary object (an implementation of the `Network` interface) to represent your network - whatever is easiest for your tests. This custom network representation will be a parameter to your test logic, and wraps the low-level `ServiceNetwork` from Kurtosis. We can see this in action in the Go implementation for a network of a fixed size [here](https://github.com/kurtosis-tech/kurtosis-go/blob/a278ad34d94e9c7968e3108cee8ed513b3eb39fd/example_impl/example_networks/fixed_size_example_network/fixed_size_example_network.go#L22).
+
+### Test Network Loader
+Just like with services, we need to tell Kurtosis how to actually initialize our custom network representation. With services, we implemented a `ServiceInitializerCore`; with networks, we'll need to implement the `NetworkLoader` interface. Before we write our implementation though, it's worth understanding how Kurtosis networks are configured. Each network has one or more **service configurations**, which serve as templates for the service instances that will comprise the network. These service configurations are defined by a configuration ID, a Docker image, a `ServiceInitializerCore` implementation, and a `ServiceAvailabilityChecker` implementation. If a network is composed of only one type of service then the network only needs one configuration; if a network is made up of many different types of services then it will need many configurations. By way of example, the Go implementation for our fixed-sized network from above can be found [here](https://github.com/kurtosis-tech/kurtosis-go/blob/a278ad34d94e9c7968e3108cee8ed513b3eb39fd/example_impl/example_networks/fixed_size_example_network/fixed_size_example_network.go#L44).
```

Comment:
@mieubrisse this is an interesting parallel I never thought of before...is it worth renaming NetworkLoader to NetworkInitializerCore, or ServiceInitializerCore to ServiceLoader, to make this parallel more obvious?

Reply:
yep, thought about this too - I've contemplated doing it, but it'd be an API break for Ava so I've thought the benefit wouldn't be worth the price. given that I'm going to be refactoring things and that's break the API anyways though, I might go ahead and do it